packages/ember-routing/lib/system/route.js

```diff
@@ -2149,6 +2142,27 @@ function buildRenderOptions(route, namePassed, isDefaultRender, name, options) {
     template: template
   };
 
+  let Component;
+  if (isEnabled('ember-routing-routable-components')) {
+    let componentName = options && options.component || namePassed && name || route.componentName || name;
+    let componentLookup = route.container.lookup('component-lookup:main');
+    Component = componentLookup.lookupFactory(componentName);
+    let isGlimmerComponent = Component && Component.proto().isGlimmerComponent;
```

Comment:
I'd prefer to do `Component.isGlimmerComponentFactory` (and do `Ember.GlimmerComponent.reopenClass({ isGlimmerComponentFactory: true});`) over `Component.proto()`.